src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeParameterInfo.cs

```diff
@@ -509,12 +509,12 @@ public override object[] GetCustomAttributes(bool inherit)
 
         public override object[] GetCustomAttributes(Type attributeType!!, bool inherit)
         {
-            if (MdToken.IsNullToken(m_tkParamDef))
-                return Array.Empty<object>();
-
             if (attributeType.UnderlyingSystemType is not RuntimeType attributeRuntimeType)
                 throw new ArgumentException(SR.Arg_MustBeType, nameof(attributeType));
 
+            if (MdToken.IsNullToken(m_tkParamDef))
+                return CustomAttribute.CreateAttributeArrayHelper(attributeRuntimeType, 0);
```

Comment:
This change was required to prevent a failure in Microsoft.CSharp.RuntimeBinder.Tests.AccessTests.PublicMethod().

I feel a bit weird exposing the CreateAttributeArrayHelper method as internal, but this is really the exact logic we want here. It makes this branch consistent with other GetCustomAttribute methods which all allow the consumer to cast the array to the requested attribute type.

Moving this after the argument check presumably has some slight perf cost but ultimately feels more correct (and unless we change CreateAttributeArrayHelper to take `Type` instead of `RuntimeType` we need the check to happen first regardless).

Reply:
Ok. Do we need a test to directly verify the behavior of ParamerInfo.GetCustomAttributes for this case?

Reply:
Added!